### PR TITLE
new style of danger button

### PR DESCRIPTION
**Pull request by ddcat1115**

### CODE REVIEW

File: components/style/themes/default.less [hunk 1]
```diff
@@ -108,8 +108,11 @@
 @btn-default-bg         : #fff;
 @btn-default-border     : @border-color-base;
 
-@btn-danger-color      : #fff;
-@btn-danger-bg         : @red-6;
+@btn-danger-color      : @red-6;
+@btn-danger-bg         : @background-color-base;
+@btn-danger-border     : @border-color-base;
+@btn-danger-hover-color: #fff;
+@btn-danger-active-bg  : @red-7;
```
Comment (afc163) [inline components/style/themes/default.less]: 感觉不需要这么多变量，后两个可以去掉。

File: components/style/themes/default.less [hunk 2]
```diff
@@ -108,8 +108,9 @@
 @btn-default-bg         : #fff;
 @btn-default-border     : @border-color-base;
 
-@btn-danger-color      : #fff;
-@btn-danger-bg         : @red-6;
+@btn-danger-color      : @red-6;
```
Comment (kappa-gooner) [inline components/style/themes/default.less]: I've a suggestion here (Which I would've raised as a separate PR if I hadn't seen this one).
Can you this variable to `@error-color`? 
https://github.com/ant-design/ant-design/blob/master/components/style/themes/default.less#L17

That way you only need to override error-color to get the desired color on the danger button. 
Makes sense? @afc163 @benjycui

Comment (afc163) [inline components/style/themes/default.less]: Yes, you are right. @ddcat1115

Comment (afc163) [inline components/style/themes/default.less]: Yes, you are right. @ddcat1115

**Pull request merged**